content/04.workflows-and-software.md

```diff
@@ -80,36 +79,52 @@ Analysis workflows commonly rely on multiple software packages to generate final
-Container-based software installation via docker and singularity is common for production-level workflows.
+Unfortunately, this system limits software version transparency and does not lend itself well to exploring new workflows and software, as researchers do not have permission to install software themselves.
+To meet this need, most workflow managers integrate with software management systems that handle software installation, management, and packaging, alleviating problems that arise from complex dependencies and facilitating documentation of software versions.
+Software management systems range from lightweight systems that manage only the software and its dependencies, to heavyweight systems that control for all aspects of the runtime and operating system, ensuring 100% reproducibility of results across computational platforms and time.
+
+On the lightweight end, the conda package manager has emerged as a leading software management solution for research workflows (**Figure @fig:conda_figure_ABC**).
+Conda handles both cluster permission and version conflict issues with a user-based software environment system, and features a straightforward "recipe" system which simplifies the process of making new software installable (including simple management of versions and updates).
+These features have led to widespread adoption within the bioinformatic community: packages for new software become quickly available, and can be installed easily across platforms.
+However, conda does not attempt to isolate software installations or standardize the runtime environment, meaning installations will not be completely reproducible over time.
+Heavyweight software management systems package not only the software of interest, but also the runtime environment information, with the goal of ensuring perfect reproducibility in software installation over time.
+Tools such as singularity and docker [@doi:10.1016/j.cels.2018.03.014; @doi:10.1038/s41592-018-0046-7; @doi:10.1371/journal.pone.0177459; @url:https://dl.acm.org/doi/10.5555/2600239.2600241] wrap software environments in "containers" that capture and reproduce the runtime environment information.
+Container-based management is particularly useful for systems where some dependencies may not be installable by lightweight managers. However, software installation within these containers can be limited by similar reproducibility issues, including changes in dependency installations over time.
+"Functional package managers" such as GNU Guix and Nix strictly require all dependency and configuration details be encoded within each software package, providing the most comprehensively reproducible installations.
+These have begun to be integrated into some bioinformatic tools, but have a steeper learning curve for independent use [@doi:10.1093/gigascience/giy123].
```

Comment:
```suggestion
These have begun to be integrated into some bioinformatic tools [@doi:10.1093/gigascience/giy123], but have a steeper learning curve for independent use.
In addition, standard installation of these managers requires system-wide installation permissions,  requiring assistance from system administrators on most high-performance computing systems.
```

Reply:
@luizirber @camillescott accurate / ok wording?

Reply:
it's correct, I think. But "requiring assistance" doesn't fully capture the horror of the situation, which involves policy level support for such things. I'm not sure it's worth iterating on - or at least I can't come up with a better phrasing ;) - but for me, it's definitely a blocker when root needs to do something!

Reply:
i know, i just can't think of a better way to put it!